source/level/ecs/systems/physics/PhysicsSystem.cpp

```diff
@@ -303,67 +299,62 @@ void PhysicsSystem::moveBody(Physics &physics, AABB &body, vec &pixelsToMove)
-                    break;
-                case right: pixelsToMove.x--;
-                case none:
-                    break;
+                case up:    pixelsToMove.y--; break;
+                case down:  pixelsToMove.y++; break;
+                case left:  pixelsToMove.x++; break;
+                case right: pixelsToMove.x--; break;
+                case none: break;
             }
         } // move is not possible:
         else if (toDo == Move::left || toDo == Move::right)  pixelsToMove.x = 0; // cant move horizontally anymore.
         else                                                 pixelsToMove.y = 0; // cant move vertically anymore.
     }
 }
 
-void PhysicsSystem::preventFallingThroughPolyPlatform(Physics &p, AABB &aabb)
+void PhysicsSystem::preventFallingThroughPolyPlatform(Physics &physics, AABB &aabb)
 {
     bool wasAbovePlatform =
-            p.prevTouched.abovePolyPlatformEntity != entt::null
-            && !p.touches.polyPlatform
-            && p.prevTouched.abovePolyPlatformEntity != p.touches.abovePolyPlatformEntity;
+            physics.prevTouched.abovePolyPlatformEntity != entt::null
+            && !physics.touches.polyPlatform
+            && physics.prevTouched.abovePolyPlatformEntity != physics.touches.abovePolyPlatformEntity;
 
-    bool wasOnPlatform = p.prevTouched.polyPlatform && !p.touches.polyPlatform;
+    bool wasOnPlatform = physics.prevTouched.polyPlatform && !physics.touches.polyPlatform;
 
     if (!wasAbovePlatform && !wasOnPlatform)
         return;
 
-    // entity MIGHT have fallen through polyPlatform
 
-    auto platformEntity = wasAbovePlatform ? p.prevTouched.abovePolyPlatformEntity : p.prevTouched.polyPlatformEntity;
+    // entity MIGHT have fallen through polyPlatform
+    auto platformEntity = wasAbovePlatform ? physics.prevTouched.abovePolyPlatformEntity : physics.prevTouched.polyPlatformEntity;
 
-    PolyPlatform *platform = room->entities.try_get<PolyPlatform>(platformEntity);
-    Polyline *line = room->entities.try_get<Polyline>(platformEntity);
-    AABB *platformAABB = room->entities.try_get<AABB>(platformEntity);
+    PolyPlatform *platform  = room->entities.try_get<PolyPlatform>(platformEntity);
+    Polyline *line          = room->entities.try_get<Polyline>(platformEntity);
+    AABB *platformAABB      = room->entities.try_get<AABB>(platformEntity);
 
-    if (!platform || !line || !platformAABB)
-        return; // platform doesn't exist anymore
+    bool
+        platformDoesNotExist        = !platform || !line || !platformAABB,
+        fallTroughIntended          = platform->allowFallThrough && physics.ignorePolyPlatforms,
+        entityWalkedOfThePlatform   = aabb.center.x <= platformAABB->bottomLeft().x || aabb.center.x > platformAABB->bottomRight().x;
 
-    if (platform->allowFallThrough && p.ignorePolyPlatforms)
-        return; // fall through was intended
+    if (platformDoesNotExist || fallTroughIntended || entityWalkedOfThePlatform)
+        return;
```

Comment:
@timostrating dit gaat extreem hard kapot als `platformDoesNotExist` true blijkt te zijn.
Je vraagt namelijk members op van `platform` en `platformAABB` die in dat geval NULL zijn.

```suggestion
    if (!platform || !line || !platformAABB)
        return; // platform doesn't exist anymore

    bool
        fallTroughIntended          = platform->allowFallThrough && physics.ignorePolyPlatforms,
        entityWalkedOfThePlatform   = aabb.center.x <= platformAABB->bottomLeft().x || aabb.center.x > platformAABB->bottomRight().x;
        
    if (fallTroughIntended || entityWalkedOfThePlatform)
        return;
```